### PR TITLE
fix(GhostRoleAntagSpawner): Create a new mind before transfering

### LINKED ISSUE #30258: antag selection spawners reuse the mind
## Description
regular ghost role spawners create a brand new mind, antag spawners seem to reuse the previous one leading to funnies:
- round end summary shows blatantly wrong names for antags
- loneop with a job real
- maybe objectives staying??? not sure

### LINKED ISSUE #30351: space dragon traitor in custody
i dont know how
image

### PR DESCRIPTION
**Pull request by Brandon-Huu**
## About the PR

Creates a new mind for players who take ghost role with the component ``GhostRoleAntagSpawner``

## Why / Balance

Resolves #30351
Resolves https://github.com/space-wizards/space-station-14/issues/29866 
Resolves #30258

image

## Technical details

Possible side effect: LoneOps no longer take the appearance of the person who took the role?
## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

### CODE REVIEW

File: Content.Server/Antag/AntagSelectionSystem.cs [hunk 1]
```diff
@@ -329,11 +329,9 @@ public void MakeAntag(Entity<AntagSelectionComponent> ent, ICommonSession? sessi
         if (session != null)
         {
             var curMind = session.GetMind();
```
Comment (EmoGarbage404) [inline Content.Server/Antag/AntagSelectionSystem.cs:331]: remove this if you're just going to make a new one

**Comment by Brandon-Huu** [comment]
There are more issues that this solves, need to find them

**Comment by Brandon-Huu** [comment]
what?

**Comment by Brandon-Huu** [comment]
Minor skill issue

**Comment by Brandon-Huu** [comment]
I blame Emo, it was building before their suggestion.

**Comment by Brandon-Huu** [comment]
Does this solve https://github.com/space-wizards/space-station-14/issues/29866 ?

**Comment by Dutch-VanDerLinde** [comment]
> Does this solve #29866 ?

yes

**Comment by deltanedas** [comment]
#30258

**Comment by slarticodefast** [comment]
Should fix #28214 as well

**Pull request merged**